xwiki-commons-core/xwiki-commons-text/src/main/java/org/xwiki/text/internal/FileSystemStoreUtils.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.text.internal;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.lang3.CharUtils;
+
+/**
+ * Various file system module utils.
+ * 
+ * @version $Id$
+ * @since 10.1RC1
+ */
+public final class FileSystemStoreUtils
+{
+    /**
+     * The standard role hint used by most filesystem store implementations.
+     */
+    public static final String HINT = "file";
```

Comment:
Feels like something you moved? I don't see where the HINT is used in your changes? And the since is wrong?

Reply:
Indeed, I moved the file without paying too much attention to that, good catch. Now this hint is used quite a lot on the `xwiki-platform` side, so I'm not sure what to do. Move it into a separate class? This is for example used in https://github.com/xwiki/xwiki-platform/blob/7dee366a67352adc78da6aef6bc67c867818414a/xwiki-platform-core/xwiki-platform-store/xwiki-platform-store-filesystem-oldcore/src/main/java/org/xwiki/store/legacy/store/internal/FilesystemAttachmentStore.java#L231

Reply:
It's internal only, so if you change the imports in xwiki-platform to use that file from that module I'd say it's ok. You just need to prepare the PR in platform for that and link it here to understand the changes I'd say.

Reply:
But it does not make much sense right now to get this metadata from `xwiki-commons-text`. Would be better to put HINT back to platform, probably somewhere in `xwiki-platform-store-filesystem`.

Reply:
I've analyzed a bit how the `encode` method of this class is used in `xwiki-platform` and I came to the conclusion that moving and re-using this class seems wrong. This `encode` method is solely used for moving attachments that were previously stored with this kind of encoding in a migration. This means that we must not change it in any way, or the migration would fail. Even if we should notice that the encoding causes problems in certain scenarios, we wouldn't change it. While we need stable encoding here, too, I believe it's a bad idea to re-use this class as here, we would change the implementation if we noticed any problems.